tasks/main.yml

```diff
@@ -58,11 +58,15 @@
   template:
     src: server.hcl.j2
     dest: "{{ nomad_config_dir }}/server.hcl"
+  when:
+    - _nomad_node_server
```

Comment:
Would you mind adding a bool filter here?
```yaml
when:
  - _nomad_node_server | bool
```

It's an internal variable, so not that important, but in the rare circumstance someone want's to override this with `-e` this will be handy.